### ISSUE TITLE
False positive: Pylint doesn't deduce non-None-ness from ternary condition

### ISSUE THREAD

**Issue by inducer** [pylint-dev/pylint#4608]
### Steps to reproduce

Given a file `a.py`:
```python
import random

if random.choice([0, 1]):
    x = None
else:
    x = 15

y = (-x if x is not None else None)
```

I get
```
a.py:8:5: E1130: bad operand type for unary -: NoneType (invalid-unary-operand-type)
```

### Expected behavior

It's clear that the `-x` won't be executed if `x` is None. IMO the warning is a false positive as a result.

### pylint --version output

Result of `pylint --version` output:

```
pylint 2.8.2
astroid 2.5.6
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110]
```

cc @majosm

**Comment by Pierre-Sassoulas** [comment]
astroid 4.0 got None comprehension and fixed this.
